frontend/app/routes/protected/person-case/name-info.tsx

```diff
@@ -0,0 +1,226 @@
+          v.maxLength(maxLength, t('protected:name-info.first-name-error.max-length-error')),
+          v.regex(REGEX_PATTERNS.NON_DIGIT, t('protected:name-info.first-name-error.format-error')),
+        ),
+        middleName: v.optional(
+          v.pipe(
+            v.string(t('protected:name-info.middle-name-error.required-error')),
+            v.trim(),
+            v.maxLength(maxLength, t('protected:name-info.middle-name-error.max-length-error')),
+            v.regex(REGEX_PATTERNS.NON_DIGIT, t('protected:name-info.middle-name-error.format-error')),
+          ),
+        ),
+        lastName: v.pipe(
+          v.string(t('protected:name-info.last-name-error.required-error')),
+          v.trim(),
+          v.nonEmpty(t('protected:name-info.last-name-error.required-error')),
+          v.maxLength(maxLength, t('protected:name-info.last-name-error.max-length-error')),
+          v.regex(REGEX_PATTERNS.NON_DIGIT, t('protected:name-info.last-name-error.format-error')),
+        ),
+        docTypes: v.pipe(
+          v.array(v.string(), t('protected:name-info.supporting-error.required-error')),
+          docsRequired
+            ? v.nonEmpty(t('protected:name-info.supporting-error.required-error'))
+            : v.empty(t('protected:name-info.supporting-error.invalid-error')),
+          v.checkItems((item) => VALID_DOC_TYPES.includes(item), t('protected:name-info.supporting-error.invalid-error')),
+        ),
+      }) satisfies v.GenericSchema<NameInfoSessionData>;
+
+      const input = {
+        firstName: formData.get('first-name') as string,
+        middleName: formData.get('middle-name')?.toString().trim() || undefined,
+        lastName: formData.get('last-name') as string,
+        docTypes: (docsRequired ? formData.getAll('doc-type') : []) as string[],
+      } satisfies Partial<NameInfoSessionData>;
+
+      const parseResult = v.safeParse(schema, input, { lang });
+
+      if (!parseResult.success) {
+        return data({ errors: v.flatten<typeof schema>(parseResult.issues).nested }, { status: 400 });
+      }
+
+      context.session.inPersonSINCase ??= {};
+      context.session.inPersonSINCase.nameInfo = parseResult.output;
+
+      throw i18nRedirect('routes/protected/request.tsx', request);
+    }
+
+    default: {
+      throw new AppError(`Unrecognized action: ${action}`);
+    }
+  }
+}
+
+export default function PrimaryDocs({ loaderData, actionData, params }: Route.ComponentProps) {
+  const { t } = useTranslation(handle.i18nNamespace);
+  const [requireDoc, setRequireDoc] = useState<boolean>(false);
+  const fetcher = useFetcher<Info['actionData']>();
+  const isSubmitting = fetcher.state !== 'idle';
+  const errors = fetcher.data?.errors;
+  const requireOptions = [
+    { children: t('gcweb:input-option.yes'), value: 'Yes', onClick: () => setRequireDoc(true) },
+    { children: t('gcweb:input-option.no'), value: 'No', onClick: () => setRequireDoc(false) },
+  ];
```

Comment:
You have to default with session value and set state when radio value change.

```suggestion
  const [requireDoc, setRequireDoc] = useState<boolean | undefined>(loaderData.defaultFormValues?.supportingDocumentsRequired);
  const fetcher = useFetcher<Info['actionData']>();
  const isSubmitting = fetcher.state !== 'idle';
  const errors = fetcher.data?.errors;
  
  function handleRequireDocChanged(event: ChangeEvent<HTMLInputElement>) {
    setRequireDoc(event.target.value === REQUIRE_OPTIONS.yes);
  }

  const requireOptions: InputRadiosProps['options'] = [
    {
      children: t('gcweb:input-option.yes'),
      value: REQUIRE_OPTIONS.yes,
      defaultChecked: requireDoc === true,
      onChange: handleRequireDocChanged,
    },
    {
      children: t('gcweb:input-option.no'),
      value: REQUIRE_OPTIONS.no,
      defaultChecked: requireDoc === false,
      onChange: handleRequireDocChanged,
    },
  ];
```